app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx

```diff
@@ -83,15 +86,27 @@ export default function CompetingStep({
       ));
     },
     onSuccess: (data) => {
-      dispatch(setMessage('registrations.flash.updated', 'positive'));
       queryClient.setQueryData(
         ['registration', competitionInfo.id, user.id],
         {
           ...data.registration,
           payment: registration.payment,
         },
       );
-      nextStep();
+      // Going from pending -> Cancelled
+      if (data.registration.competing.registration_status === 'cancelled') {
+        dispatch(setMessage('competitions.registration_v2.register.registration_status.cancelled', 'positive'));
+        nextStep({ toStart: true });
```

Comment:
I could return here to next one level deeper?

Reply:
Seems you already made up your mind :smile: